src/web4/handler.rs

```diff
@@ -0,0 +1,331 @@
+                    image = community.logo_url;
+                }
+                redirect_path = format!("devhub.near/widget/app?page={}&handle={}", page, handle);
+                initial_props_json = json!({"page": page, "handle": handle}).to_string();
+            }
+            "proposal" => {
+                let id_string = path_parts[2];
+                let id_option = id_string.parse::<u32>();
+                if id_option.is_ok() {
+                    let id = id_option.unwrap();
+                    let proposal_option = contract.proposals.get(id.into());
+                    if proposal_option.is_some() {
+                        let proposal: Proposal = Proposal::from(proposal_option.unwrap());
+                        let proposal_body = proposal.snapshot.body.latest_version();
+                        title = html_escape::encode_text(proposal_body.name.as_str()).to_string();
+                        description =
+                            html_escape::encode_text(proposal_body.summary.as_str()).to_string();
+                    }
+                }
+                redirect_path = format!("devhub.near/widget/app?page={}&id={}", page, id_string);
+                initial_props_json = json!({"page": page, "id": id_string}).to_string();
+            }
+            _ => {}
+        }
+    }
+
+    let body = format!(
+        r#"<!DOCTYPE html>
+<html>
+<head>
+    <title>{title}</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width,initial-scale=1">
+    <meta property="og:url" content="{url}" />
+    <meta property="og:type" content="website" />
+    <meta property="og:title" content="{title}" />
+    <meta property="og:description" content="{description}" />
+    <meta property="og:image" content="{image}" />
+
+    <meta name="twitter:card" content="summary_large_image">
+    <meta name="twitter:title" content="{title}">
+    <meta name="twitter:description" content="{description}">
+    <meta name="twitter:image" content="{image}">
+    <script src="https://ipfs.web4.near.page/ipfs/bafybeic6aeztkdlthx5uwehltxmn5i6owm47b7b2jxbbpwmydv2mwxdfca/main.794b6347ae264789bc61.bundle.js"></script>
+    <script src="https://ipfs.web4.near.page/ipfs/bafybeic6aeztkdlthx5uwehltxmn5i6owm47b7b2jxbbpwmydv2mwxdfca/runtime.25b143da327a5371509f.bundle.js"></script>
+</head>
+<body>
+    <div class="container">
+        <nav class="navbar navbar-expand-lg navbar-light bg-light">
+            <div class="navbar-brand"><img src="https://i.near.social/magic/large/https://near.social/magic/img/account/devhub.near" style="width: 64px" /></div>
+            <ul class="navbar-nav mr-auto">
+                <li class="nav-item"><a href="https://near.org/{redirect_path}" class="nav-link">near.org</a></li>
+                <li class="nav-item"><a href="https://near.social/{redirect_path}" class="nav-link">near.social</a></li>
+            </ul>
```

Comment:
The current design is not clear. We need to make the action item more clear. Many mobile apps have a standard UI for it:

![IMG_7BFF98F59CAC-1](https://github.com/NEAR-DevHub/neardevhub-contract/assets/304265/c742ee51-fda1-4ce3-a771-419da9cb13d5)

![IMG_C268E5E04096-1](https://github.com/NEAR-DevHub/neardevhub-contract/assets/304265/82cb528a-87e9-4b33-ad84-ce4432a7555d)

If you get stuck, ask Sasha / Ori to help you with the design of this bar.

Reply:
@ori-near would you or Sasha provide some UI design suggestion for the buttons to open in near.org/near.social that I can implement?

Reply:
@petersalomonsen Absolutely! Please find UI design here:
https://www.figma.com/file/llwJeKCuDxC89694GiQZrS/DevHub-Social-Media-Links?type=design&node-id=0%3A1&mode=design&t=yHplqngP876MHR56-1

Reply:
Implemented now @ori-near @frol